### PR TITLE
Close #345

### LINKED ISSUE #345: Clarify behaviour of existence test with OptionalBoolean function
Currently, the existence test description contradicts the intended use of OptionalBoolean functions. We need to be careful if we want to make a special case for OptionalBoolean functions and not open the door to any boolean value being treated similarly.

### PR DESCRIPTION
**Pull request by cabo**
Add the necessary internal references.

### CODE REVIEW

File: draft-ietf-jsonpath-base.md [hunk 1]
```diff
@@ -1069,7 +1069,10 @@ the current node is selected if and only if the expression yields true.
 
 The current node is accessible via the current node identifier `@`.
 
-An existence expression may test the result of a function expression (see {{fnex}}).
+An existence expression either tests the existence of a node
+designated by an embedded query (see {{extest}}) or tests whether the
+result of a function expression of type `Boolean` or `OptionalBoolean`
+(see {{fnex}}) is `true`.
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:1076]: The above sentence doesn't cover existence tests of function return values of type `OptionalNodes` or one of its subtypes. I think the solution to that is to beef up the extest section to cover such function return values.

Comment (cabo) [reply draft-ietf-jsonpath-base.md:1076]: Yes.  Will update PR later today.

File: draft-ietf-jsonpath-base.md [hunk 2]
```diff
@@ -1069,7 +1069,10 @@ the current node is selected if and only if the expression yields true.
...
```
Comment (glyn) [inline draft-ietf-jsonpath-base.md:1075]: I don't think of functions expressions of type `Boolean` as existence expressions. We test the return value, but existence is moot. I'd prefer to raise them to another kind of basic-expr.

`OptionalBoolean` function expressions involve both an existence test and, if the return value is not `Nothing`, consideration of the return value.

I wonder if we should try to tease this out in the spec?

**Review by glyn — CHANGES_REQUESTED** [review]
A missing case and an improvement to consider.

File: draft-ietf-jsonpath-base.md (hunk 2 above)
Comment (cabo) [reply draft-ietf-jsonpath-base.md:1075]: I'd like to fix the ABNF to be structured along more logical lines.  This is really #344; I think in this PR we should make sure we have all the material that we then reshuffle.

File: draft-ietf-jsonpath-base.md (hunk 1 above)
Comment (cabo) [reply draft-ietf-jsonpath-base.md:1095]: ... and I resolved the conflict by merging main.

**Review by glyn — APPROVED** [review]
Ok, let's refactor the ABNF later.

**Pull request merged**